pkgs/os-specific/linux/plymouth/default.nix

```diff
@@ -1,95 +1,96 @@
 { lib
 , stdenv
-, fetchpatch
 , fetchFromGitLab
+, meson
 , pkg-config
-, autoreconfHook
+, libpng
+, udev
+, pango
+, gtk3
+, libdrm
+, libevdev
+, libxkbcommon
+, xorg
 , libxslt
+, ninja
 , docbook-xsl-nons
-, gettext
-, gtk3
+, ncurses
 , systemd
-, pango
-, cairo
-, libdrm
 }:
 
 stdenv.mkDerivation rec {
   pname = "plymouth";
-  version = "unstable-2021-10-18";
-
-  outputs = [
-    "out"
-    "dev"
-  ];
+  version = "unstable-2023-02-22";
 
   src = fetchFromGitLab {
     domain = "gitlab.freedesktop.org";
-    owner = "plymouth";
-    repo = "plymouth";
-    rev = "18363cd887dbfe7e82a2f4cc1a49ef9513919142";
-    sha256 = "sha256-+AP4ALOFdYFt/8MDXjMaHptkogCwK1iXKuza1zfMaws=";
+    owner = pname;
+    repo = pname;
+    rev = "7302b49da8b98d01ab82d23e5be74146065cffd8";
+    hash = "sha256-PgVJezSUd/E3o1YigVto5KmQlflONrmKspOmAHSc5p4=";
   };
 
   nativeBuildInputs = [
-    autoreconfHook
-    docbook-xsl-nons
-    gettext
-    libxslt
+    meson
     pkg-config
+    ninja
+    docbook-xsl-nons
   ];
 
   buildInputs = [
-    cairo
+    libpng
+    udev
+    pango
     gtk3
     libdrm
-    pango
+    libevdev
+    libxkbcommon
+    xorg.xkeyboardconfig
+    libxslt
+    ncurses
     systemd
   ];
 
-  postPatch = ''
-    sed -i \
-      -e "s#plymouthplugindir=.*#plymouthplugindir=/etc/plymouth/plugins/#" \
-      -e "s#plymouththemedir=.*#plymouththemedir=/etc/plymouth/themes#" \
-      -e "s#plymouthpolicydir=.*#plymouthpolicydir=/etc/plymouth/#" \
-      -e "s#plymouthconfdir=.*#plymouthconfdir=/etc/plymouth/#" \
-      configure.ac
-  '';
-
-  configurePlatforms = [ "host" ];
-
-  configureFlags = [
-    "--enable-documentation"
-    "--enable-drm"
-    "--enable-gtk"
-    "--enable-pango"
-    "--enable-systemd-integration"
-    "--enable-tracing"
-    "--localstatedir=/var"
+  mesonFlags = [
     "--sysconfdir=/etc"
-    "--with-background-color=0x000000"
-    "--with-background-end-color-stop=0x000000"
-    "--with-background-start-color-stop=0x000000"
-    "--with-logo=/etc/plymouth/logo.png"
-    "--with-release-file=/etc/os-release"
-    "--with-runtimedir=/run"
-    "--with-systemdunitdir=${placeholder "out"}/etc/systemd/system"
-    "--without-rhgb-compat-link"
-    "--without-system-root-install"
-    "ac_cv_path_SYSTEMD_ASK_PASSWORD_AGENT=${lib.getBin systemd}/bin/systemd-tty-ask-password-agent"
+    "-Dlogo=/etc/plymouth/logo.png"
+    "-Dbackground-color=0x000000"
+    "-Dbackground-start-color-stop=0x000000"
+    "-Dbackground-end-color-stop=0x000000"
+    "-Drelease-file=/etc/os-release"
+    "-Dtracing=true"
+    "-Dupstart-monitoring=true"
```

Comment:
Some of these are probably useless. upstart was, IIRC a service manager used by Ubuntu in ancient past before systemd. And I believe someone mentioned in the past on one of the  the previous PRs that there were some issues with freetype. But it looks like it is now [enabled by default](https://gitlab.freedesktop.org/plymouth/plymouth/-/blob/7302b49da8b98d01ab82d23e5be74146065cffd8/meson_options.txt#L68) so probably it is fine now and the flag is redundant.